src/hypothesis/strategies.py

```diff
@@ -660,13 +671,37 @@ def characters(whitelist_categories=None, blacklist_categories=None,
                 min_codepoint, max_codepoint
             )
         )
+    if all((whitelist_characters is not None,
+            min_codepoint is None,
+            max_codepoint is None,
+            whitelist_categories is None,
+            blacklist_categories is None,
+            )):
+        raise InvalidArgument(
+            'Cannot have just whitelist_characters=%r alone, '
+            'it would have no effect. Perhaps you want sampled_from()' % (
+                whitelist_characters,
+            )
+        )
+    if (
+        whitelist_characters is not None and
+        blacklist_characters is not None and
+        set(whitelist_characters).intersection(set(whitelist_characters))
```

Comment:
Ugh. Right. Oops. I totally should have caught that in review, sorry. @moreati, would you mind fixing this or would you like me to?